File: Backend/PixelDread/Models/Post.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;

 namespace PixelDread.Models
{
    public class Post
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public string? Name { get; set; }

        public DateTime? CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }

        [Required]
        public bool Visibility { get; set; }

        public IdentityUser? User { get; set; }
        public string? UserId { get; set; }

        public int? CategoryId { get; set; }

        public Category? Category { get; set; }

        public List<PostTag> PostTags { get; set; } = new List<PostTag>();

        public List<PostArticle> PostArticles { get; set; } = new List<PostArticle>();

        public int? OGDataId { get; set; }

        public OGData? OGData { get; set; }
    }
}

File: Frontend/src/types/post.ts
import { ArticleUnion } from "./articles";

export interface Category {
  id: number;
  name: string;
  default?: boolean;
}

export interface PostArticle {
  postId: number;
  articleId: number;
  articleType: number; 
  order: number;
  article: ArticleUnion;
}

export interface PostTag {
  postId: number;
  tagId: number;
  tag: {
    id: number;
    name: string;
  };
}

export interface Post {
  id: number;
  name: string;
  createdAt?: string;
  updatedAt?: string;
  visibility: boolean;
  userId: string;
  categoryId?: number;
  category?: Category;
  postArticles: PostArticle[];
  postTags?: PostTag[];
  ogData?: {
    title?: string;
    description?: string;
    fileInformationsId?: number;
    slug?: string;
  };
}

